pkg/didmethod/httpbinding/resolver.go

```diff
@@ -0,0 +1,137 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package httpbinding
+
+import (
+	"crypto/tls"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	errors "golang.org/x/xerrors"
+)
+
+// HTTPClientOpts holds options for the HTTP transport implementation of DIDResolver
+// it has a required http.Client instance and optional basePath which is appended to the remove DID Resolver URL
+type HTTPClientOpts struct {
+	client   *http.Client
+	basePath string
+}
+
+// HTTPOptSetter is an outbound HTTP transport option
+type HTTPOptSetter func(opts *HTTPClientOpts)
+
+// WithHTTPClientInstance option is for creating an Outbound HTTP transport using an http.Client instance
+func WithHTTPClientInstance(client *http.Client) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client = client
+	}
+}
+
+// WithHTTPClientTimeout option is for creating an Outbound HTTP transport using a client timeout value
+func WithHTTPClientTimeout(timeout time.Duration) HTTPOptSetter {
+	return func(opts *HTTPClientOpts) {
+		opts.client.Timeout = timeout
```

Comment:
Panic if `WithHTTPClientInstance` is not called first.

I wonder if we should consider substituting this `WithXXX` options pattern with the one discussed [here](https://github.com/hyperledger/aries-framework-go/pull/120#discussion_r314558349)

@troyronda @Baha-sk

Reply:
We should use functional options for the New constructor. A struct is too brittle.

Reply:
see https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis for a longer explanation. also an example of this pattern is widely used in fabric-sdk-go and grpc.

Reply:
Opened #127

Reply:
I left functional options approach in this PR.

Reply:
Also setting `DefaultClient` at the initialization before applying functional options for safety reasons.